ui.apps/src/main/content/jcr_root/apps/asset-share-commons/clientlibs/clientlib-site/js/cart/event-handlers.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Asset Share Commons
+ *
+ * Copyright [2017]  Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*global jQuery: false, AssetShare: false, handleButtonsUpdateOnDetails, handleBulkCartButtonsUpdate, toggleCartButtons, flipAction  */
+jQuery((function($, ns, cart) {
+	"use strict";
+	var element,
+		ADD_TO_CART = "add-to-cart",
+		REMOVE_FROM_CART = "remove-from-cart",
+		EVENT_DETAILS_PAGE_LOAD = "asset-share-commons.details.load",
```

Comment:
I wonder if we should just have a "asset-share-commons.page.load" event that we just fire whenever the DOM is ready for any page the main library is loaded. This add remove functionality should (and i think is in this PR) implemented in such a way its agnostic to what page its been added to. Just thinking ahead if we introduce some new page type... this feature should just look for the add/remove cart button asset-share-ids and toggle accordingly. WDYT?

Reply:
Hi @davidjgonzalez ,
Introduced "asset-share-commons.page.load" event. This event gets fired whenever DOM is ready for all pages and checks if the page has add/remove cart button asset-share-ids and toggles accordingly. Kindly have a look